libs/libc/stdio/lib_libvsprintf.c

```diff
@@ -878,8 +878,32 @@ static int vsprintf_internal(FAR struct lib_outstream_s *stream,
                   ndigs += 1;
                 }
 
-              putc(ndigs, stream);
-              putc('0' + exp, stream);
+              /* Parse the ndigs if the value of it bigger than '9' */
+
+              while (1)
```

Comment:
Maybe better to

```suggestion
              while (ndigits > 0)
```
Or something like this. Just from some static alalyzers point of view